libs/common/util/src/lib/object.ts

```diff
@@ -215,3 +215,19 @@ export function reverseMap<K extends string, V extends string>(
     Object.entries(obj).map(([k, v]) => [v, k])
   ) as Record<V, K>
 }
+
+export function shallowCompareObj<T extends Record<string, any>>(
+  obj1: T,
+  obj2: T
+) {
+  const keys1 = Object.keys(obj1)
+  const keys2 = Object.keys(obj2)
+
+  // Check if the objects have the same number of keys
+  if (keys1.length !== keys2.length) return false
+
+  // Check if all keys and their values are the same
+  for (const key of keys1) if (obj1[key] !== obj2[key]) return false
```

Comment:
I assume not really an issue, but isnt there a bug for objs like
```typescript
const obj1 = {
  'key0': undefined
  'key1': 'a'
}
const obj2 = {
  'keyNot0': undefined
  'key1': 'a'
}
```

Reply:
technically they are the same object (at least in the JSON representation), I don't think its a big problem.

Reply:
Depends on what you use the keys for but ye, not a problem in current usage

Reply:
I get ya, in this case even if an entry is undefined, it technically defines the "key" in this object
![image](https://github.com/user-attachments/assets/6899a63e-e588-48fa-8ed2-1b776740711b)